### PR TITLE
[WIP] Upgrade Knex to latest version and swap unsupported sqlite3 fork

### PR DESCRIPTION
**Pull request by derrickmehaffy**
Signed-off-by: Derrick Mehaffy <derrickmehaffy@gmail.com>

Upgrades Knex to latest version (v1.0.3) and swaps out mapbox's node-sqlite3 for Microsoft's @vscode/sqlite3 instead

Also adds two new dialects from Knex:
- better-sqlite3
- mysql2

I've set the default in this PR to make better-sqlite3 the default instead of sqlite, we should consider dropping normal sqlite3 in the future.
mysql2 is required for MySQL v8+ and it's new auth mechanism, we should consider dropping mysql dialect in the future.

### CODE REVIEW

File: packages/generators/app/lib/utils/db-client-dependencies.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 const sqlClientModule = {
   mysql: { mysql: '2.18.1' },
   postgres: { pg: '8.6.0' },
-  sqlite: { sqlite3: '5.0.2' },
+  sqlite: { '@vscode/sqlite3': '5.0.7' },
```
Comment (alexandrebodin) [inline packages/generators/app/lib/utils/db-client-dependencies.js:6]: It doesn't appear like the original package has gone unmaintained and they are not redirecting to the microsoft one.  What's the reasoning here ?

Comment (derrickmehaffy) [reply packages/generators/app/lib/utils/db-client-dependencies.js:6]: > It doesn't appear like the original package has gone unmaintained and they are not redirecting to the microsoft one. What's the reasoning here ?

There are various security issues in the vulnerabilities and mapbox isn't responding, Knex themselves moved to using the vscode fork and if we don't use it, Knex will throw errors if the sqlite dialect is used without that package installed.

Comment (alexandrebodin) [reply packages/generators/app/lib/utils/db-client-dependencies.js:6]: Ok I see, I'm not 100% comfortable ditching a package that was provided by a team for so long without thinking about it a bit more. I would like to see if there are plans for them to maintain the package in the long term or if they are letting Microsoft take over. Is there an issue on their repo with a conversation around that topic ?

 Right now the knex upgrade seems to be break all the tests anyway. Once that's fixed I guess we will have to consider it more

Comment (derrickmehaffy) [reply packages/generators/app/lib/utils/db-client-dependencies.js:6]: I think that time long past https://github.com/knex/knex/issues/4858

Honestly if we want to go a better route we can drop both of these and move to better-sqlite3 which is a full fork with more features and has a proven history.

**Comment by derrickmehaffy** [comment]
Needs more testing

**Comment by derrickmehaffy** [comment]
super-seeded by: https://github.com/strapi/strapi/pull/12918

(Of course alex is a boss)

**Pull request closed without merging**